RevenueCatUI/Templates/V2/ViewHelpers/Shape.swift

```diff
@@ -59,35 +60,52 @@ struct ShapeModifier: ViewModifier {
 
     var border: BorderInfo?
     var shape: Shape
+    var shadow: ShadowModifier.ShadowInfo?
+    var background: BackgroundStyle?
 
-    init(border: BorderInfo? = nil, shape: Shape?) {
+    init(border: BorderInfo? = nil,
+         shape: Shape?,
+         shadow: ShadowModifier.ShadowInfo? = nil,
+         background: BackgroundStyle? = nil) {
         self.border = border
         self.shape = shape ?? .rectangle(nil)
+        self.shadow = shadow
+        self.background = background
     }
 
     func body(content: Content) -> some View {
         switch self.shape {
-        case .rectangle(let radiuses):
+        case .rectangle(let radiusInfo):
+            let shape = self.effectiveRectangleShape(radiusInfo: radiusInfo)
+            let effectiveShape = shape ?? Rectangle().eraseToAnyInsettableShape()
             content
-                .conditionalClipShape(topLeft: radiuses?.topLeft,
-                                      topRight: radiuses?.topRight,
-                                      bottomLeft: radiuses?.bottomLeft,
-                                      bottomRight: radiuses?.bottomRight)
-                .conditionalOverlay(color: self.border?.color,
-                                    width: self.border?.width,
-                                    topLeft: radiuses?.topLeft,
-                                    topRight: radiuses?.topRight,
-                                    bottomLeft: radiuses?.bottomLeft,
-                                    bottomRight: radiuses?.bottomRight)
+                .backgroundStyle(background)
+                // We want to clip only in case there is a non-Rectangle shape
+                // or if there's a border, otherwise we let the background color
+                // extend behind the safe areas
```

Comment:
I noticed this when it changed in an Emerge snapshot.

When the shape is a rectangle without rounded corners and there is no border, if we don't clip the background, it can extend behind the safe areas. This might be useful for the root view and for the sticky footer view so it felt important to keep the behavior and document it.

<img width="100" alt="Screenshot 2025-01-03 at 22 08 19" src="https://github.com/user-attachments/assets/3a3d9e0d-05b1-4f91-922a-3f8e5bcaa01c" />
<img width="100" alt="Screenshot 2025-01-03 at 22 08 51" src="https://github.com/user-attachments/assets/2a7df848-854f-44de-a42e-ecbc2d594473" />